### ISSUE TITLE
is jit.rawkernel(device=True) appears to work with tuple return values

### ISSUE THREAD

**Issue by truher** [cupy/cupy#6959]
### Description

is this supposed to work?  it appears to work, if the function is called from another jit.rawkernel, and it's nice that it works.  i could make an example for the docs if you want.

**Comment by leofang** [comment]
Interesting, thanks for sharing. 

@asi1024 correct me if I am wrong, I don't think it's intentional but is indeed a nice-to-have feature that's accidentally enabled (as a byproduct) in #6668.

**Comment by truher** [comment]
heh.  that's cool.  it might be nice to mention that tuples *do not* work as return values *without* device=True.  that caused a bit of consternation for me this morning, as i was seeking to test a "device" function directly.

**Comment by leofang** [comment]
We definitely cannot return any value (not just tuple) from a CUDA kernel (all kernels are annotated as `__global__ void ...` in CUDA C++), but device functions (with `__device__` identifier in CUDA) that can be called by kernels can have an arbitrary return type, as long as the CuPy JIT compiler knows how to interpret it, and it seems `tuple` is now one of such cases.

**Comment by leofang** [comment]
> that's accidentally enabled (as a byproduct) in https://github.com/cupy/cupy/pull/6668.

I am probably wrong, this should already be supported much earlier than the linked PR.

**Comment by leofang** [comment]
btw forgot to say (sorry for the distraction), any help to improve code/documentation is more than welcomed!

**Comment by truher** [comment]
oh ha ha, yes of course, all are void, i was only using tuple so that's all i saw, my mistake.
i'll work up an example and send you a PR.

**Comment by asi1024** [comment]
@truher Yes, it is supposed to work! We welcome PRs to add examples to `examples/jit`, and documentation to `docs/source/user_guide/kernel.rst`!
